dace/transformation/dataflow/stream_transient.py

```diff
@@ -226,36 +228,48 @@ def apply(self, sdfg: SDFG):
             node_a=map_exit,
             node_b=outer_map_exit)
 
-        # Initialize transient to zero in case of summation, or with an identity
-        # map/tasklet in other cases
-        memlet = graph.in_edges(data_node)[0].data
-        if detect_reduction_type(memlet.wcr) == dtypes.ReductionType.Sum:
-            data_node.setzero = True
-        else:
-            if self.identity is None:
-                warnings.warn('AccumulateTransient did not properly initialize'
-                              'newly-created transient!')
-                return
-
-            init_state = sdfg.add_state_before(graph)
-            desc = sdfg.arrays[array]
-            if desc.total_size == 1:
-                # Add initialization as a tasklet
-                t = init_state.add_tasklet('acctrans_init', {}, {'out'},
-                                           'out = %s' % self.identity)
-                w = init_state.add_write(array)
-                init_state.add_edge(t, 'out', w, None, dace.Memlet(array))
-            else:
-                # Add initialization as a map
-                init_state.add_mapped_tasklet(
-                    'acctrans_init', {
-                        '_o%d' % i: '0:%s' % symstr(d)
-                        for i, d in enumerate(desc.shape)
-                    }, {},
-                    'out = %s' % self.identity, {
-                        'out':
-                        dace.Memlet.simple(
-                            '_out', ','.join(
-                                ['_o%d' % i for i in range(desc.shape)]))
-                    },
-                    external_edges=True)
+        if self.identity is None:
+            warnings.warn('AccumulateTransient did not properly initialize'
+                          'newly-created transient!')
+            return
+
+        from dace.transformation.helpers import nest_state_subgraph
+        from dace.sdfg.graph import SubgraphView
+        from dace.sdfg.state import SDFGState
+
+
+        sdfg_state: SDFGState = sdfg.node(self.state_id)
+
+        map_entry = sdfg_state.entry_node(map_exit)
+
+        dtypes_key = ('compiler', 'default_data_types')
+        old_dtype = dace.Config.get(*dtypes_key)
+        dace.Config.set(*dtypes_key, value='C')
+
+        from dace.sdfg.nodes import NestedSDFG
+
+        nested_sdfg: NestedSDFG = nest_state_subgraph(sdfg=sdfg, state=sdfg_state, subgraph=SubgraphView(sdfg_state, [map_entry, tasklet, map_exit]))
```

Comment:
This will not work for any map with more than one tasklet. Please use `state.scope_subgraph(map_entry)` as the subgraph instead.